diff --git a/omi/firmware/devkit/CMakeLists.txt b/omi/firmware/devkit/CMakeLists.txt
index 2dfea18..4b2eb3c 100644
--- a/omi/firmware/devkit/CMakeLists.txt
+++ b/omi/firmware/devkit/CMakeLists.txt
@@ -36,6 +36,7 @@ target_sources(app PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../app/src/wr_time_sync.c
     ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../app/src/wr_storage_service.c
     ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../app/src/wr_boot_cmd.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../app/src/wr_wdt_feeder.c
 )
 target_include_directories(app PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../app/include
